client/src/scene/SceneGameArena.ts

```diff
@@ -130,12 +135,17 @@ export class SceneGameArena extends Phaser.Scene {
     update(time: number, delta: number) {
         this.frameTimeElapsed += delta;
 
+        if (this.trade == null && this.gameState.playerId != null) {
```

Comment:
I think yes, the trade portion needs to be moved up. The reason I moved the controlsUI up is because it needed a `playerId` to identify players from spectators. Now that we are sending `initPlayer` to scene waiting room (meaning that scene game arena will have the playerID ready), there is no need to initialize components in update.

Speaking of `playerID` for player identification, I don't see that the TradeUI is in need of a playerID, which sounds like every one including spectator can see it right now. Correct me if I missed anything, but if that's the case maybe we should fix that in this PR (or another one).  

cc @AnimeshKC

Reply:
@c25vdw if we do remove the playerID check, then everyone will be able to view the currently ongoing trades. That is fine but it would mean we would have to hide the "shift key" portion of the UI so that spectators are not presented with that information.
So in either case, we will need to perform the playerID check

Reply:
Trade UI view for spectator will be handled in a different PR after the spectator view is resolved. Right now, enabling the Trade UI for non-players has strange effects on the spectator view.